File: lib/domain/model/playback_state.dart
import 'track.dart';

enum PlaybackRepeatMode { off, one, all }

class PlaybackState {
  final Track? currentTrack;
  final bool isPlaying;
  final Duration position;
  final Duration duration;
  final PlaybackRepeatMode repeatMode;
  final bool shuffleEnabled;
  final List<Track> queue;
  final int queueIndex;

  const PlaybackState({
    this.currentTrack,
    this.isPlaying = false,
    this.position = Duration.zero,
    this.duration = Duration.zero,
    this.repeatMode = PlaybackRepeatMode.off,
    this.shuffleEnabled = false,
    this.queue = const [],
    this.queueIndex = 0,
  });

  PlaybackState copyWith({
    Track? currentTrack,
    bool? isPlaying,
    Duration? position,
    Duration? duration,
    PlaybackRepeatMode? repeatMode,
    bool? shuffleEnabled,
    List<Track>? queue,
    int? queueIndex,
  }) {
    return PlaybackState(
      currentTrack: currentTrack ?? this.currentTrack,
      isPlaying: isPlaying ?? this.isPlaying,
      position: position ?? this.position,
      duration: duration ?? this.duration,
      repeatMode: repeatMode ?? this.repeatMode,
      shuffleEnabled: shuffleEnabled ?? this.shuffleEnabled,
      queue: queue ?? this.queue,
      queueIndex: queueIndex ?? this.queueIndex,
    );
  }
}

File: app/src/main/java/com/soundflow/app/domain/model/PlaybackState.kt
package com.soundflow.app.domain.model

enum class RepeatMode {
    OFF,
    ONE,
    ALL
}

data class PlaybackState(
    val currentTrack: Track? = null,
    val isPlaying: Boolean = false,
    val position: Long = 0L,
    val duration: Long = 0L,
    val shuffleEnabled: Boolean = false,
    val repeatMode: RepeatMode = RepeatMode.OFF,
    val queue: List<Track> = emptyList(),
    val queueIndex: Int = 0
)
